src/main/java/io/cryostat/net/web/http/api/v2/TargetsPostHandler.java

```diff
@@ -171,13 +187,27 @@ public IntermediateResponse<ServiceRef> handle(RequestParameters params) throws
             cryostatAnnotations.put(AnnotationKey.REALM, CustomTargetPlatformClient.REALM);
             serviceRef.setCryostatAnnotations(cryostatAnnotations);
 
-            boolean v = customTargetPlatformClient.addTarget(serviceRef);
-            if (!v) {
-                throw new ApiException(400, "Duplicate connectUrl");
+            if (!dryRun) {
+                boolean v = customTargetPlatformClient.addTarget(serviceRef);
+                if (!v) {
+                    throw new ApiException(400, "Duplicate connectUrl");
+                }
             }
             return new IntermediateResponse<ServiceRef>().body(serviceRef);
         } catch (JvmIdGetException e) {
-            throw new ApiException(404, "Couldn't connect to target: " + e.getTarget());
+            if (AbstractAuthenticatedRequestHandler.isJmxAuthFailure(e)) {
+                throw new ApiException(406, "Credentials Not Provided/Not Acceptable", e);
```

Comment:
I don't think using 406 is the best idea. Usually, I believe status 406 is reserved for throwing an error when the the content-type the user requests does not match what the endpoint can accept. Maybe using 401 Unauthorized is better?

Reply:
Just wondering if status code 401 confict with other places? Like cases of missing`Authentication: `?

Reply:
How about an unused 4xx status code that is closer to 427? How about 432?

Reply:
Using 420 for now.